src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -259,26 +260,49 @@ private void openBrowserImpl(String uri, String screen) {
       devToolsInstance.openBrowserAndConnect(uri, screen);
     }
     else {
-      @Nullable final OSProcessHandler handler =
-        isBazel(project) ? getProcessHandlerForBazel() : getProcessHandlerForPub();
-
-      if (handler != null) {
-        // TODO(devoncarew) Add a Task.Backgroundable here; "Starting devtools..."
-
-        // start the server
-        DevToolsInstance.startServer(handler, instance -> {
-          devToolsInstance = instance;
+      // For internal users, we can connect to the DevTools server started by flutter daemon. For external users, the flutter daemon has an
+      // older version of DevTools, so we launch the server using `pub global run` instead.
+      if (isBazel(project)) {
+        final Optional<FlutterApp> first =
+          FlutterApp.allFromProjectProcess(project).stream().filter((FlutterApp app) -> app.getProject() == project).findFirst();
+
+        if (!first.isPresent()) {
+          return;
+        }
 
-          devToolsInstance.openBrowserAndConnect(uri, screen);
-        }, () -> {
-          // Listen for closing; null out the devToolsInstance.
-          devToolsInstance = null;
+        first.get().serveDevTools().thenAccept((Pair<String, Integer> hostAndPort) -> {
+          if (hostAndPort == null) {
+            @Nullable final OSProcessHandler handler = getProcessHandlerForBazel();
+            startDevToolsServerAndConnect(handler, uri, screen);
+          } else {
+            devToolsInstance = new DevToolsInstance(hostAndPort.first, hostAndPort.second);
+            devToolsInstance.openBrowserAndConnect(uri, screen);
+          }
         });
+      } else {
+        @Nullable final OSProcessHandler handler = getProcessHandlerForPub();
+        startDevToolsServerAndConnect(handler, uri, screen);
       }
-      else {
-        // TODO(devoncarew): We should provide feedback to callers that the open browser call failed.
+    }
+  }
+
+  private void startDevToolsServerAndConnect(OSProcessHandler handler, String uri, String screen) {
+    if (handler != null) {
+      // TODO(devoncarew) Add a Task.Backgroundable here; "Starting devtools..."
```

Comment:
use your username for new TODOs

Reply:
I didn't want to put my name here since I just moved this block of code without changes, do you think this is enough of a change where I should take ownership of the TODO?

Reply:
you did the right thing. I saw the lower one was a move but I didn't scroll up high enough to spot this one.